### ISSUE TITLE
[BUG] Duplicate posts in GetPostResponse.posts

### ISSUE THREAD

**Issue by boscojwho** [Athlon007/Leomard#41]
**Describe the bug**
There are duplicate `PostView.post` instances in GetPostResponse.posts array. This is causing SwiftUI List to emit warning in console, and might(?) be causing some weird scrolling behaviour.
- Not sure if Lemmy API is returning duplicate posts with the same `post.id` or something else.

**To Reproduce**
1. Open Feed or Community list view.
2. Scroll down until you see SwiftUI warning about Post occurring multiple times in collection. 

**Expected behavior**
- List should not contain duplicate posts (uniqued by `post.id`).

**Information:**
- Version 0.1 (1)

**Comment by Athlon007** [comment]
I suppose it's happening when user scrolls down, and by the time he reaches the end, a new post appears on Lemmy, which pushes last post(s) to the next page. I guess the only way to avoid it is to check if the new added post is not in the posts array.

**Comment by Athlon007** [comment]
Addressed in #45

**Comment by boscojwho** [comment]
@Athlon007 I'm still seeing SwiftUI errors complaining about duplicate data in views. I haven't checked if it's this same issue or something else entirely. It's causing issues when scrolling, but I will investigate =)

**Comment by boscojwho** [comment]
For example, this was in Community view when scrolling through posts list:
```
ForEach<Array<PostView>, PostView, TupleView<(ModifiedContent<ModifiedContent<PostUIView, _AppearanceActionModifier>, _FlexFrameLayout>, ModifiedContent<Spacer, _FrameLayout>)>>: the ID PostView(post: Leomard.Post(id: 1853043, name: "[Self Promo] My Lemmy client, Bean is now available on TestFlight", body: Optional("Hey 👋 I\'ve been working on a new Lemmy client called Bean for the last few weeks and I\'ve put out an early beta.\n\nIt\'s missing some features at the moment but I\'m working to get it feature complete by early next week and would appreciate some feedback as I develop!\n\nYou can find out more over at [!bean@lemmy.world](https://lemmy.world/c/bean) but here are a few screenshots.\n\nLet me know what you think if you do check it out.\n\n\n![](https://lemmy.world/pictrs/image/f2d1a543-739c-44c8-b7a1-9f74744d9dfa.png)\n![](https://lemmy.world/pictrs/image/0dcaceb4-846a-4bac-ad59-5466e4605c81.png)\n![](https://lemmy.world/pictrs/image/11e7a3a8-fd7b-437b-88c0-8a5f8c329278.png)\n![](https://lemmy.world/pictrs/image/dadc9751-eace-463c-a542-da5627cbd3da.png)\n![](https://lemmy.world/pictrs/image/2831d382-9c29-4678-a6c3-67723ad2da48.png)\n![](https://lemmy.world/pictrs/image/a568a91b-0be9-431b-bd2c-0a845a3ce05b.png)"), creatorId: 368954, communityId: 1256, removed: false, locked: false, published: 2023-07-20 11:31:11 +0000, updated: Optional(2023-07-21 09:53:33 +0000), deleted: false, nsfw: false, apId: "https://lemmy.world/post/1853043", local: true, languageId: 37, featuredCommunity: false, featuredLocal: false, url: Optional("https://testflight.apple.com/join/de5zt3wH"), thumbnailUrl: Optional("https://lemmy.world/pictrs/image/7eb5cedd-a0ef-4fee-b719-a2e311f0fb95.png"), embedTitle: Optional("Join the Bean for Lemmy beta"), embedDescription: Optional("Available on iOS"), embedVideoUrl: nil), creator: Leomard.Person(id: 368954, name: "steve228uk", displayName: Optional("steve228uk"), avatar: Optional("https://lemmy.world/pictrs/image/9911e69b-a9fd-4d33-9bae-d74c2a670381.png"), banner: Optional("https://lemmy.world/pictrs/image/30d92d66-8a98-4d03-b7c7-cb70a4d62a36.jpeg"), banned: false, published: 2023-06-23 11:27:09 +0000, actorId: "https://lemmy.world/u/steve228uk", bio: Optional("Hello I\'m Steve"), local: true, deleted: false, matrixUserId: Optional("@steve228uk:example.com"), admin: false, botAccount: false, instanceId: 1, updated: nil), community: Leomard.Community(id: 1256, name: "apple_enthusiast", title: "Apple", banner: Optional("https://lemmy.world/pictrs/image/51344035-2373-464a-a5a5-eb5f13e7dbb5.jpeg"), description: Optional("##### Welcome            \nto the largest Apple community on Lemmy. This is the place where we talk about everything Apple, from iOS to the exciting upcoming Apple Vision Pro. Feel free to join the discussion!\n\n\n##### Rules:  \n\n1. No NSFW Content  \n1. No Hate Speech or Personal Attacks  \n1. No Ads / Spamming   \n   1. Self promotion is considered an ad\n\n[Lemmy Code of Conduct](https://join-lemmy.org/docs/code_of_conduct.html)\n\n##### Communities of Interest:  \nApple Hardware      \n[Apple TV](https://lemmy.world/c/appletv)      \n[Apple Watch  ](https://lemmy.world/c/applewatch)      \n[iPad ](https://lemmy.world/c/ipad)       \n[iPhone ](https://lemmy.world/c/iphone_community)      \n[Mac](https://lemmy.world/c/mac_community)     \n[Vintage Apple](https://lemmy.world/c/vintage_apple)\n\nApple Software        \n[iOS](https://lemmy.world/c/ios_community)      \n[iPadOS](https://lemmy.world/c/ipados)     \n[macOS](https://lemmy.world/c/macos)     \n[tvOS](https://lemmy.world/c/tvos)   \n[watchOS](https://lemmy.world/c/watchos)     \n[Shortcuts](https://lemmy.world/c/shortcuts)  \n[Xcode](https://lemmy.world/c/xcode)            "), removed: false, published: 2023-06-09 19:17:21 +0000, updated: Optional(2023-07-19 08:07:17 +0000), deleted: false, nsfw: false, actorId: "https://lemmy.world/c/apple_enthusiast", local: true, icon: Optional("https://lemmy.world/pictrs/image/78c808b2-acf4-492d-8776-fff2ad3c6a69.jpeg"), hidden: false, postingRestrictedToMods: false, instanceId: 1), creatorBannedFromCommunity: false, counts: Leomard.PostAggregates(id: 384018, postId: 1853043, comments: 128, score: 455, upvotes: 470, downvotes: 15, published: 2023-07-20 11:31:11 +0000, newestCommentTimeNecro: 2023-07-22 03:35:00 +0000, newestCommentTime: 2023-07-23 17:56:10 +0000, featuredCommunity: false, featuredLocal: false, hotRank: 2, hotRankActive: 3), subscribed: "Subscribed", saved: false, read: false, creatorBlocked: false, unreadComments: 128, myVote: nil) occurs multiple times within the collection, this will give undefined results!
```

**Comment by Athlon007** [comment]
Gah, I forgot to filter out already present posts in Community View. Fixed in #76
